test/persona_client_test.js

```diff
@@ -1670,10 +1683,624 @@ describe("Persona Client Test Suite", function(){
+                redis_host:"localhost",
+                redis_port:6379,
+                redis_db:0,
+                enable_debug: false
+            });
+
+            personaClient.getScopesForUser("guid",{},function(err,data) {
+                assert(err!=null);
+                err.should.be.a.String;
+                err.should.equal("guid and token are required strings");
+                assert(data===null);
+                done();
+            });
+        });
+
+        it("should throw an error if guid is not valid", function(done) {
+            var personaClient = persona.createClient({
+                persona_host:"persona",
+                persona_port:80,
+                persona_scheme:"http",
+                persona_oauth_route:"/oauth/tokens/",
+                redis_host:"localhost",
+                redis_port:6379,
+                redis_db:0,
+                enable_debug: false
+            });
+
+            personaClient.obtainToken("primate","bananas",function(err,data1) {
+                personaClient.getScopesForUser("guid",data1.access_token,function(err,data) {
+                    assert(err!=null);
+                    err.should.be.a.String;
+                    err.should.equal("getScopesForUser failed with status code 404");
+                    assert(data===null);
+                    done();
+                });
+            });
+        });
+
+        it("should return scopes if guid is valid", function(done) {
+            global.http = require('http');
+            var requestStub = sinon.stub(http, 'request');
+
+            var data = {scope:['fdgNy6QWGmIAl7BRjEsFtk','tdc:app:access','tdc:player:access']};
+            var expected = data.scope;
+            var response = new PassThrough();
+            response.statusCode = 200;
+            response.write(JSON.stringify(data));
+            response.end();
+            var request = new PassThrough();
+
+            var personaClient = persona.createClient({
+                persona_host:"persona",
+                persona_port:80,
+                persona_scheme:"http",
+                persona_oauth_route:"/oauth/tokens/",
+                redis_host:"localhost",
+                redis_port:6379,
+                redis_db:0,
+                enable_debug: false
+            });
+            requestStub.callsArgWith(1, response).returns(request);
+
+            // stub out the call to get obtainToken by providing a fake redisClient that returns known data
+            var now = new Date().getTime() / 1000;
+            var reply = {access_token: 'thisisatoken', expires_at: now + 1000};
+
+            sinon.stub(personaClient.redisClient, "get").callsArgWith(1,null,JSON.stringify(reply));
+
+            personaClient.obtainToken("primate","bananas",function(err,data1) {
+                personaClient.getScopesForUser('guid_does_exist', data1.access_token, function(err, data){
+                    assert(err===null);

```

Comment:
`==` and can we have some spacing between operators please? I think it improves legibility.